apps/target-app/app.py

```diff
@@ -63,68 +65,73 @@ def filtered_data() -> pd.DataFrame:
-        with ui.layout_columns():
-            with ui.card():
-                ui.card_header("Model Metrics")
-
-                @render_plotly
-                def metric():
-                    df_value = df()
-                    df_filtered = df_value[df_value["account"] == input.account()]
-                    if input.metric() == "ROC Curve":
-                        return plot_auc_curve(
-                            df_filtered, "is_electronics", "training_score"
-                        )
-                    else:
-                        return plot_precision_recall_curve(
-                            df_filtered, "is_electronics", "training_score"
-                        )
-
-                ui.input_select(
-                    "metric",
-                    "Metric",
-                    choices=["ROC Curve", "Precision-Recall"],
-                )
-            with ui.card():
-                ui.card_header("Training Scores")
-
-                @render_plotly
-                def score_dist():
-                    df_value = df()
-                    df_filtered = df_value[df_value["account"] == input.account()]
-                    return plot_score_distribution(df_filtered)
-
-        with ui.card(full_screen=True):
-            with ui.card_header():
-                "Data"
-            with ui.popover(title="Download"):
+ui.nav_spacer()
+
+with ui.nav_panel("Training Dashboard"):
+    with ui.layout_columns():
+        with ui.card():
+            ui.card_header("Model Metrics")
+
+            @render_plotly
+            def metric():
+                df_value = df()
+                df_filtered = df_value[df_value["account"] == input.account()]
+                if input.metric() == "ROC Curve":
+                    return plot_auc_curve(
+                        df_filtered, "is_electronics", "training_score"
+                    )
+                else:
+                    return plot_precision_recall_curve(
+                        df_filtered, "is_electronics", "training_score"
+                    )
+
+            ui.input_select(
+                "metric",
+                "Metric",
+                choices=["ROC Curve", "Precision-Recall"],
+            )
+        with ui.card():
+            ui.card_header("Training Scores")
+
+            @render_plotly
+            def score_dist():
+                df_value = df()
+                df_filtered = df_value[df_value["account"] == input.account()]
+                return plot_score_distribution(df_filtered)
+
+    with ui.card(full_screen=True):
+        with ui.card_header():
+            "Data"
+            with ui.popover(
+                title="Download",
+                class_="d-inline-block pull-right",
+            ):
                 fa.icon_svg("download")
 
-                @render.download()
-                def download_data(filename="scores_data.csv"):
+                @render.download(filename="scores_data.csv")
+                def download_data():
```

Comment:
`filename` needs to be passed to `@render.download()`, not the function it's decorating. Without this, the downloaded file is just named `download_data` or something generic.